### ISSUE TITLE
Linux GridView scroll with trackpad is sometimes reversed.

### ISSUE THREAD

**Issue by LimaneGaya** [flutter/flutter#150799]
### Steps to reproduce

1. Scroll with trackpad in one direction. (happens more often when touching the trackpad with one finger first then adding the second then scrolling).
2. Repeat until noticing the behavior.

### Expected results

Should scroll in direction set in os.

### Actual results

Scrolls in opposite direction.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(home: Scaffold(body: MyWidget())));
}

const sliverdelegate = SliverGridDelegateWithMaxCrossAxisExtent(
    maxCrossAxisExtent: 300, childAspectRatio: 1);

class MyWidget extends StatelessWidget {
  const MyWidget({super.key});

  @override
  Widget build(BuildContext context) {
    return GridView.builder(
      gridDelegate: sliverdelegate,
      itemCount: Colors.primaries.length,
      itemBuilder: (context, i) => Container(color: Colors.primaries[i]),
    );
  }
}
```

</details>

### Flutter Doctor output

Does not work

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel master, 3.23.0-13.0.pre.326, on KDE neon 6.0 6.5.0-41-generic, locale fr_FR.UTF-8)
    • Flutter version 3.23.0-13.0.pre.326 on channel master at /home/gaya/software/flutter_dev
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e5a35f458f (il y a 77 minutes), 2024-06-25 11:27:38 -0700
    • Engine revision 62e0b5f9c3
    • Dart version 3.5.0 (build 3.5.0-297.0.dev)
    • DevTools version 2.36.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/gaya/software/androidsdk/
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /home/gaya/software/androidsdk/
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+9-Ubuntu-122.04.1)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 14.0.0-1ubuntu1.1
    • cmake version 3.29.0
    • ninja version 1.10.1
    • pkg-config version 1.8.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • KDE neon 6.0 6.5.0-41-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 126.0.6478.126

[✓] Network resources
    • All expected network resources are available.

```

</details>

Works in 

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel beta, 3.23.0-0.1.pre, on KDE neon 6.0 6.5.0-41-generic, locale fr_FR.UTF-8)
    • Flutter version 3.23.0-0.1.pre on channel beta at /home/gaya/software/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2feea7a407 (il y a 3 semaines), 2024-06-06 10:19:10 +0700
    • Engine revision bb10c54666
    • Dart version 3.5.0 (build 3.5.0-180.3.beta)
    • DevTools version 2.36.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/gaya/software/androidsdk/
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /home/gaya/software/androidsdk/
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+9-Ubuntu-122.04.1)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 14.0.0-1ubuntu1.1
    • cmake version 3.29.0
    • ninja version 1.10.1
    • pkg-config version 1.8.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • KDE neon 6.0 6.5.0-41-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 126.0.6478.126

[✓] Network resources
    • All expected network resources are available
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @LimaneGaya, is this only on linux desktop or is the web affected too? 

What scroll direction is set in your OS? Can you share a recording of the issue to better help us reproduce the issue? 

Thank you

**Comment by LimaneGaya** [comment]
@danagbemava-nc yes i can only reproduce this on linux, not on web and i can't try on other platforms.
The scroll setting is set to natural scrolling.

left is the master branch and right is beta:
in both i keep scrolling up the same way i mentioned (touching trackpad with one finger first then the other then scrolling). 
As shown in the recording sometimes in the master branch scrolls downward (the scroll direction get inverted)

https://github.com/flutter/flutter/assets/36268750/ccf5aaa6-8678-44bc-8d30-ed5655736656

**Comment by danagbemava-nc** [comment]
Hi @LimaneGaya, how long does it typically take for you to notice this? I've been testing for a few minutes but I can't seem to reproduce the issue. I tested on the latest master (as at now) [`3.23.0-13.0.pre.357`] on ubuntu. 

Does this only happen with natural scrolling enabled? Can you also check with a master channel newer than `3.23.0-13.0.pre.326` to see if this still reproduces?

**Comment by LimaneGaya** [comment]
Thank you @danagbemava-nc for investigating the issue. 

This seems to be fixed by #150275.
